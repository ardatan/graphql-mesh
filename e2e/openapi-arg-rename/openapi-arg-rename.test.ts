import { createTenv } from '@e2e/tenv';

const { compose, serve, service } = createTenv(__dirname);

describe('OpenAPI Additional Resolvers', () => {
  it('composes the schema', async () => {
    const { result } = await compose({
      output: 'graphql',
      services: [await service('Wiki')],
      maskServicePorts: true,
    });

    expect(result).toMatchSnapshot();
  });
  it('should work with untouched schema', async () => {
    const { output } = await compose({
      output: 'graphql',
      services: [await service('Wiki')],
    });

    const { execute } = await serve({ supergraph: output });
    const queryResult = await execute({
      query: /* GraphQL */ `
        mutation Good {
          post_good(input: { banana: true }) {
            apple
          }
        }
      `,
    });

    expect(queryResult.data).toEqual({
      post_good: [
        {
          apple: '{"banana":true}',
        },
      ],
    });
  });

  it('should work with renamed argument', async () => {
    const { output } = await compose({
      output: 'graphql',
      services: [await service('Wiki')],
    });

    const { execute } = await serve({ supergraph: output });
    const queryResult = await execute({
      query: /* GraphQL */ `
        mutation Bad {
          post_bad(requestBody: { banana: true }) {
            apple
          }
        }
      `,
    });

    expect(queryResult.data).toEqual({
      post_bad: [
        {
          apple: '{"banana":true}',
        },
      ],
    });
  });
});
