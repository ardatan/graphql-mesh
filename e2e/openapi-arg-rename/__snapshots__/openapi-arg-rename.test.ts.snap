// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenAPI Additional Resolvers composes the schema 1`] = `
"
schema
  @link(url: "https://specs.apollo.dev/link/v1.0")
  @link(url: "https://specs.apollo.dev/join/v0.3", for: EXECUTION)
  
  
  
  
  
  @link(
  url: "https://the-guild.dev/graphql/mesh/spec/v1.0"
  import: ["@httpOperation", "@transport", "@source", "@extraSchemaDefinitionDirective"]
)
{
  query: Query
  mutation: Mutation
  
}


  directive @join__enumValue(graph: join__Graph!) repeatable on ENUM_VALUE

  directive @join__field(
    graph: join__Graph
    requires: join__FieldSet
    provides: join__FieldSet
    type: String
    external: Boolean
    override: String
    usedOverridden: Boolean
  ) repeatable on FIELD_DEFINITION | INPUT_FIELD_DEFINITION

  directive @join__graph(name: String!, url: String!) on ENUM_VALUE

  directive @join__implements(
    graph: join__Graph!
    interface: String!
  ) repeatable on OBJECT | INTERFACE

  directive @join__type(
    graph: join__Graph!
    key: join__FieldSet
    extension: Boolean! = false
    resolvable: Boolean! = true
    isInterfaceObject: Boolean! = false
  ) repeatable on OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT | SCALAR

  directive @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION

  scalar join__FieldSet


  directive @link(
    url: String
    as: String
    for: link__Purpose
    import: [link__Import]
  ) repeatable on SCHEMA

  scalar link__Import

  enum link__Purpose {
    """
    \`SECURITY\` features provide metadata necessary to securely resolve fields.
    """
    SECURITY

    """
    \`EXECUTION\` features provide metadata necessary for operation execution.
    """
    EXECUTION
  }







enum join__Graph {
  WIKI @join__graph(name: "Wiki", url: "http://localhost:<Wiki_port>") 
}

directive @httpOperation(
  subgraph: String
  path: String
  operationSpecificHeaders: [[String]]
  httpMethod: HTTPMethod
  isBinary: Boolean
  requestBaseBody: ObjMap
  queryParamArgMap: ObjMap
  queryStringOptionsByParam: ObjMap
  jsonApiFields: Boolean
  queryStringOptions: ObjMap
) repeatable on FIELD_DEFINITION

directive @transport(
  subgraph: String
  kind: String
  location: String
  headers: [[String]]
  queryStringOptions: ObjMap
  queryParams: [[String]]
) repeatable on SCHEMA

directive @source(name: String!, type: String, subgraph: String!, hoist: _HoistConfig)  repeatable on SCALAR | OBJECT | FIELD_DEFINITION | ARGUMENT_DEFINITION | INTERFACE | UNION | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

directive @extraSchemaDefinitionDirective(directives: _DirectiveExtensions)  repeatable on OBJECT

scalar ObjMap @join__type(graph: WIKI) 

scalar _HoistConfig @join__type(graph: WIKI) 

scalar _DirectiveExtensions @join__type(graph: WIKI) 

type Query @extraSchemaDefinitionDirective(
  directives: {transport: [{subgraph: "Wiki", kind: "rest", location: "http://localhost:<Wiki_port>"}]}
) @join__type(graph: WIKI)  {
  dummy: String
}

type Mutation @join__type(graph: WIKI)  {
  post_good(input: RequestBody_Input) : [ResponseBody] @httpOperation(
    subgraph: "Wiki"
    path: "/good"
    operationSpecificHeaders: [["Content-Type", "application/json"], ["accept", "application/json"]]
    httpMethod: POST
  )
  post_bad(
    requestBody: RequestBody_Input @source(name: "input", type: "RequestBody_Input", subgraph: "Wiki") 
  ): [ResponseBody] @httpOperation(
    subgraph: "Wiki"
    path: "/bad"
    operationSpecificHeaders: [["Content-Type", "application/json"], ["accept", "application/json"]]
    httpMethod: POST
  )
}

type ResponseBody @join__type(graph: WIKI)  {
  apple: String
}

enum HTTPMethod @join__type(graph: WIKI)  {
  GET @join__enumValue(graph: WIKI) 
  HEAD @join__enumValue(graph: WIKI) 
  POST @join__enumValue(graph: WIKI) 
  PUT @join__enumValue(graph: WIKI) 
  DELETE @join__enumValue(graph: WIKI) 
  CONNECT @join__enumValue(graph: WIKI) 
  OPTIONS @join__enumValue(graph: WIKI) 
  TRACE @join__enumValue(graph: WIKI) 
  PATCH @join__enumValue(graph: WIKI) 
}

input RequestBody_Input @join__type(graph: WIKI)  {
  banana: Boolean
}
    "
`;
